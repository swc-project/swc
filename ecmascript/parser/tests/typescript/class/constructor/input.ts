class C {
    constructor(x: number, y: number);
    constructor(x: string, y: string);
    constructor(x: any, y: any) {}
}
