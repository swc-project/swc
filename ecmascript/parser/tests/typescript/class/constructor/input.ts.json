{
  "type": "Script",
  "span": {
    "start": 0,
    "end": 124,
    "ctxt": 0
  },
  "body": [
    {
      "type": "ClassDeclaration",
      "identifier": {
        "type": "Identifier",
        "span": {
          "start": 6,
          "end": 7,
          "ctxt": 0
        },
        "value": "C",
        "optional": false
      },
      "declare": false,
      "span": {
        "start": 0,
        "end": 124,
        "ctxt": 0
      },
      "decorators": [],
      "body": [
        {
          "type": "Constructor",
          "span": {
            "start": 14,
            "end": 48,
            "ctxt": 0
          },
          "key": {
            "type": "Identifier",
            "span": {
              "start": 14,
              "end": 25,
              "ctxt": 0
            },
            "value": "constructor",
            "optional": false
          },
          "params": [
            {
              "type": "Parameter",
              "span": {
                "start": 26,
                "end": 35,
                "ctxt": 0
              },
              "decorators": [],
              "pat": {
                "type": "Identifier",
                "span": {
                  "start": 26,
                  "end": 35,
                  "ctxt": 0
                },
                "value": "x",
                "optional": false,
                "typeAnnotation": {
                  "type": "TsTypeAnnotation",
                  "span": {
                    "start": 27,
                    "end": 35,
                    "ctxt": 0
                  },
                  "typeAnnotation": {
                    "type": "TsKeywordType",
                    "span": {
                      "start": 29,
                      "end": 35,
                      "ctxt": 0
                    },
                    "kind": "number"
                  }
                }
              }
            },
            {
              "type": "Parameter",
              "span": {
                "start": 37,
                "end": 46,
                "ctxt": 0
              },
              "decorators": [],
              "pat": {
                "type": "Identifier",
                "span": {
                  "start": 37,
                  "end": 46,
                  "ctxt": 0
                },
                "value": "y",
                "optional": false,
                "typeAnnotation": {
                  "type": "TsTypeAnnotation",
                  "span": {
                    "start": 38,
                    "end": 46,
                    "ctxt": 0
                  },
                  "typeAnnotation": {
                    "type": "TsKeywordType",
                    "span": {
                      "start": 40,
                      "end": 46,
                      "ctxt": 0
                    },
                    "kind": "number"
                  }
                }
              }
            }
          ],
          "body": null,
          "accessibility": null,
          "isOptional": false
        },
        {
          "type": "Constructor",
          "span": {
            "start": 53,
            "end": 87,
            "ctxt": 0
          },
          "key": {
            "type": "Identifier",
            "span": {
              "start": 53,
              "end": 64,
              "ctxt": 0
            },
            "value": "constructor",
            "optional": false
          },
          "params": [
            {
              "type": "Parameter",
              "span": {
                "start": 65,
                "end": 74,
                "ctxt": 0
              },
              "decorators": [],
              "pat": {
                "type": "Identifier",
                "span": {
                  "start": 65,
                  "end": 74,
                  "ctxt": 0
                },
                "value": "x",
                "optional": false,
                "typeAnnotation": {
                  "type": "TsTypeAnnotation",
                  "span": {
                    "start": 66,
                    "end": 74,
                    "ctxt": 0
                  },
                  "typeAnnotation": {
                    "type": "TsKeywordType",
                    "span": {
                      "start": 68,
                      "end": 74,
                      "ctxt": 0
                    },
                    "kind": "string"
                  }
                }
              }
            },
            {
              "type": "Parameter",
              "span": {
                "start": 76,
                "end": 85,
                "ctxt": 0
              },
              "decorators": [],
              "pat": {
                "type": "Identifier",
                "span": {
                  "start": 76,
                  "end": 85,
                  "ctxt": 0
                },
                "value": "y",
                "optional": false,
                "typeAnnotation": {
                  "type": "TsTypeAnnotation",
                  "span": {
                    "start": 77,
                    "end": 85,
                    "ctxt": 0
                  },
                  "typeAnnotation": {
                    "type": "TsKeywordType",
                    "span": {
                      "start": 79,
                      "end": 85,
                      "ctxt": 0
                    },
                    "kind": "string"
                  }
                }
              }
            }
          ],
          "body": null,
          "accessibility": null,
          "isOptional": false
        },
        {
          "type": "Constructor",
          "span": {
            "start": 92,
            "end": 122,
            "ctxt": 0
          },
          "key": {
            "type": "Identifier",
            "span": {
              "start": 92,
              "end": 103,
              "ctxt": 0
            },
            "value": "constructor",
            "optional": false
          },
          "params": [
            {
              "type": "Parameter",
              "span": {
                "start": 104,
                "end": 110,
                "ctxt": 0
              },
              "decorators": [],
              "pat": {
                "type": "Identifier",
                "span": {
                  "start": 104,
                  "end": 110,
                  "ctxt": 0
                },
                "value": "x",
                "optional": false,
                "typeAnnotation": {
                  "type": "TsTypeAnnotation",
                  "span": {
                    "start": 105,
                    "end": 110,
                    "ctxt": 0
                  },
                  "typeAnnotation": {
                    "type": "TsKeywordType",
                    "span": {
                      "start": 107,
                      "end": 110,
                      "ctxt": 0
                    },
                    "kind": "any"
                  }
                }
              }
            },
            {
              "type": "Parameter",
              "span": {
                "start": 112,
                "end": 118,
                "ctxt": 0
              },
              "decorators": [],
              "pat": {
                "type": "Identifier",
                "span": {
                  "start": 112,
                  "end": 118,
                  "ctxt": 0
                },
                "value": "y",
                "optional": false,
                "typeAnnotation": {
                  "type": "TsTypeAnnotation",
                  "span": {
                    "start": 113,
                    "end": 118,
                    "ctxt": 0
                  },
                  "typeAnnotation": {
                    "type": "TsKeywordType",
                    "span": {
                      "start": 115,
                      "end": 118,
                      "ctxt": 0
                    },
                    "kind": "any"
                  }
                }
              }
            }
          ],
          "body": {
            "type": "BlockStatement",
            "span": {
              "start": 120,
              "end": 122,
              "ctxt": 0
            },
            "stmts": []
          },
          "accessibility": null,
          "isOptional": false
        }
      ],
      "superClass": null,
      "isAbstract": false,
      "typeParams": null,
      "superTypeParams": null,
      "implements": []
    }
  ],
  "interpreter": null
}
