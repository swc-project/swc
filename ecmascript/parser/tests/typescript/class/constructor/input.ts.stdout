Module {
    span: Span {
        lo: BytePos(
            0
        ),
        hi: BytePos(
            124
        ),
        ctxt: #0
    },
    body: [
        Stmt(
            Decl(
                Class(
                    ClassDecl {
                        ident: Ident {
                            sym: C,
                            span: Span {
                                lo: BytePos(
                                    6
                                ),
                                hi: BytePos(
                                    7
                                ),
                                ctxt: #0
                            },
                            type_ann: None,
                            optional: false
                        },
                        declare: false,
                        class: Class {
                            span: Span {
                                lo: BytePos(
                                    0
                                ),
                                hi: BytePos(
                                    124
                                ),
                                ctxt: #0
                            },
                            decorators: [],
                            body: [
                                Constructor(
                                    Constructor {
                                        span: Span {
                                            lo: BytePos(
                                                14
                                            ),
                                            hi: BytePos(
                                                48
                                            ),
                                            ctxt: #0
                                        },
                                        key: Ident(
                                            Ident {
                                                sym: constructor,
                                                span: Span {
                                                    lo: BytePos(
                                                        14
                                                    ),
                                                    hi: BytePos(
                                                        25
                                                    ),
                                                    ctxt: #0
                                                },
                                                type_ann: None,
                                                optional: false
                                            }
                                        ),
                                        params: [
                                            Pat(
                                                Ident(
                                                    Ident {
                                                        sym: x,
                                                        span: Span {
                                                            lo: BytePos(
                                                                26
                                                            ),
                                                            hi: BytePos(
                                                                27
                                                            ),
                                                            ctxt: #0
                                                        },
                                                        type_ann: Some(
                                                            TsTypeAnn {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        27
                                                                    ),
                                                                    hi: BytePos(
                                                                        35
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                type_ann: TsKeywordType(
                                                                    TsKeywordType {
                                                                        span: Span {
                                                                            lo: BytePos(
                                                                                29
                                                                            ),
                                                                            hi: BytePos(
                                                                                35
                                                                            ),
                                                                            ctxt: #0
                                                                        },
                                                                        kind: TsNumberKeyword
                                                                    }
                                                                )
                                                            }
                                                        ),
                                                        optional: false
                                                    }
                                                )
                                            ),
                                            Pat(
                                                Ident(
                                                    Ident {
                                                        sym: y,
                                                        span: Span {
                                                            lo: BytePos(
                                                                37
                                                            ),
                                                            hi: BytePos(
                                                                38
                                                            ),
                                                            ctxt: #0
                                                        },
                                                        type_ann: Some(
                                                            TsTypeAnn {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        38
                                                                    ),
                                                                    hi: BytePos(
                                                                        46
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                type_ann: TsKeywordType(
                                                                    TsKeywordType {
                                                                        span: Span {
                                                                            lo: BytePos(
                                                                                40
                                                                            ),
                                                                            hi: BytePos(
                                                                                46
                                                                            ),
                                                                            ctxt: #0
                                                                        },
                                                                        kind: TsNumberKeyword
                                                                    }
                                                                )
                                                            }
                                                        ),
                                                        optional: false
                                                    }
                                                )
                                            )
                                        ],
                                        body: None,
                                        accessibility: None,
                                        is_optional: false
                                    }
                                ),
                                Constructor(
                                    Constructor {
                                        span: Span {
                                            lo: BytePos(
                                                53
                                            ),
                                            hi: BytePos(
                                                87
                                            ),
                                            ctxt: #0
                                        },
                                        key: Ident(
                                            Ident {
                                                sym: constructor,
                                                span: Span {
                                                    lo: BytePos(
                                                        53
                                                    ),
                                                    hi: BytePos(
                                                        64
                                                    ),
                                                    ctxt: #0
                                                },
                                                type_ann: None,
                                                optional: false
                                            }
                                        ),
                                        params: [
                                            Pat(
                                                Ident(
                                                    Ident {
                                                        sym: x,
                                                        span: Span {
                                                            lo: BytePos(
                                                                65
                                                            ),
                                                            hi: BytePos(
                                                                66
                                                            ),
                                                            ctxt: #0
                                                        },
                                                        type_ann: Some(
                                                            TsTypeAnn {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        66
                                                                    ),
                                                                    hi: BytePos(
                                                                        74
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                type_ann: TsKeywordType(
                                                                    TsKeywordType {
                                                                        span: Span {
                                                                            lo: BytePos(
                                                                                68
                                                                            ),
                                                                            hi: BytePos(
                                                                                74
                                                                            ),
                                                                            ctxt: #0
                                                                        },
                                                                        kind: TsStringKeyword
                                                                    }
                                                                )
                                                            }
                                                        ),
                                                        optional: false
                                                    }
                                                )
                                            ),
                                            Pat(
                                                Ident(
                                                    Ident {
                                                        sym: y,
                                                        span: Span {
                                                            lo: BytePos(
                                                                76
                                                            ),
                                                            hi: BytePos(
                                                                77
                                                            ),
                                                            ctxt: #0
                                                        },
                                                        type_ann: Some(
                                                            TsTypeAnn {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        77
                                                                    ),
                                                                    hi: BytePos(
                                                                        85
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                type_ann: TsKeywordType(
                                                                    TsKeywordType {
                                                                        span: Span {
                                                                            lo: BytePos(
                                                                                79
                                                                            ),
                                                                            hi: BytePos(
                                                                                85
                                                                            ),
                                                                            ctxt: #0
                                                                        },
                                                                        kind: TsStringKeyword
                                                                    }
                                                                )
                                                            }
                                                        ),
                                                        optional: false
                                                    }
                                                )
                                            )
                                        ],
                                        body: None,
                                        accessibility: None,
                                        is_optional: false
                                    }
                                ),
                                Constructor(
                                    Constructor {
                                        span: Span {
                                            lo: BytePos(
                                                92
                                            ),
                                            hi: BytePos(
                                                122
                                            ),
                                            ctxt: #0
                                        },
                                        key: Ident(
                                            Ident {
                                                sym: constructor,
                                                span: Span {
                                                    lo: BytePos(
                                                        92
                                                    ),
                                                    hi: BytePos(
                                                        103
                                                    ),
                                                    ctxt: #0
                                                },
                                                type_ann: None,
                                                optional: false
                                            }
                                        ),
                                        params: [
                                            Pat(
                                                Ident(
                                                    Ident {
                                                        sym: x,
                                                        span: Span {
                                                            lo: BytePos(
                                                                104
                                                            ),
                                                            hi: BytePos(
                                                                105
                                                            ),
                                                            ctxt: #0
                                                        },
                                                        type_ann: Some(
                                                            TsTypeAnn {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        105
                                                                    ),
                                                                    hi: BytePos(
                                                                        110
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                type_ann: TsKeywordType(
                                                                    TsKeywordType {
                                                                        span: Span {
                                                                            lo: BytePos(
                                                                                107
                                                                            ),
                                                                            hi: BytePos(
                                                                                110
                                                                            ),
                                                                            ctxt: #0
                                                                        },
                                                                        kind: TsAnyKeyword
                                                                    }
                                                                )
                                                            }
                                                        ),
                                                        optional: false
                                                    }
                                                )
                                            ),
                                            Pat(
                                                Ident(
                                                    Ident {
                                                        sym: y,
                                                        span: Span {
                                                            lo: BytePos(
                                                                112
                                                            ),
                                                            hi: BytePos(
                                                                113
                                                            ),
                                                            ctxt: #0
                                                        },
                                                        type_ann: Some(
                                                            TsTypeAnn {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        113
                                                                    ),
                                                                    hi: BytePos(
                                                                        118
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                type_ann: TsKeywordType(
                                                                    TsKeywordType {
                                                                        span: Span {
                                                                            lo: BytePos(
                                                                                115
                                                                            ),
                                                                            hi: BytePos(
                                                                                118
                                                                            ),
                                                                            ctxt: #0
                                                                        },
                                                                        kind: TsAnyKeyword
                                                                    }
                                                                )
                                                            }
                                                        ),
                                                        optional: false
                                                    }
                                                )
                                            )
                                        ],
                                        body: Some(
                                            BlockStmt {
                                                span: Span {
                                                    lo: BytePos(
                                                        120
                                                    ),
                                                    hi: BytePos(
                                                        122
                                                    ),
                                                    ctxt: #0
                                                },
                                                stmts: []
                                            }
                                        ),
                                        accessibility: None,
                                        is_optional: false
                                    }
                                )
                            ],
                            super_class: None,
                            is_abstract: false,
                            type_params: None,
                            super_type_params: None,
                            implements: []
                        }
                    }
                )
            )
        )
    ],
    shebang: None
}
